### PR TITLE
Controllers: Fix segfault when features can't be created

### PR DESCRIPTION
**Pull request by garbear**
## Description

During development, some feature definitions were missing from the controller XML definition, causing features to not be created. However, we pre-allocate features in the array, which means that until the feature is created, the array holds a nullptr.

Later, we use the array without checking for nullptrs, leading to a segfault. Instead of checking for nullptrs, though, we just ensure that no nullptrs enter the array in the first place.

## Motivation and context

In production, features won't be missing, but a segfault due to mistakes when editing XML is never fun.

## How has this been tested?

Tested as part of the Player Manager based on 20 alpha 2 release.

If I get time I'll write up the exact steps to reproduce.

## What is the effect on users?

No effect, because features aren't missing from XML in production.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: xbmc/input/joysticks/generic/InputHandling.cpp [hunk 1]
```diff
@@ -102,13 +102,19 @@ bool CInputHandling::OnDigitalMotion(const CDriverPrimitive& source, bool bPress
...
-
-    if (!feature)
-      feature = FeaturePtr(CreateFeature(featureName));
+    auto it = m_features.find(featureName);
+    if (it == m_features.end())
+    {
+      FeaturePtr feature(CreateFeature(featureName));
+      if (feature)
+      {
+        m_features.insert({featureName, std::move(feature)});
```
Comment (lrusak) [inline xbmc/input/joysticks/generic/InputHandling.cpp:111]: `std::map<>::insert` returns a `std::pair<iterator, bool>` Maybe you could skip the extra `find`?

```
it = m_features.insert({featureName, std::move(feature)}).first;
```

Comment (garbear) [reply xbmc/input/joysticks/generic/InputHandling.cpp:111]: rockstar. will update tomorrow

Comment (howie-f) [reply xbmc/input/joysticks/generic/InputHandling.cpp:111]: or
```cpp
std::tie(it, std::ignore) = m_features.insert({featureName, std::move(feature)});
```
if you do not want the extra bool

Comment (garbear) [reply xbmc/input/joysticks/generic/InputHandling.cpp]: thanks, updated

**Review by lrusak — APPROVED** [review]
Nice!

**Comment by lrusak** [comment]
jenkinst build this please

**Comment by lrusak** [comment]
jenkins build this please

**Pull request merged**